pulsar-client-cpp/include/pulsar/Authentication.h

```diff
@@ -19,65 +19,68 @@
 #ifndef PULSAR_AUTHENTICATION_H_
 #define PULSAR_AUTHENTICATION_H_
 
-#include <vector>
-#include <string>
-#include <map>
-#include <boost/shared_ptr.hpp>
 #include <pulsar/Result.h>
 #include <boost/make_shared.hpp>
+#include <boost/shared_ptr.hpp>
+#include <map>
+#include <string>
+#include <vector>
 
 #pragma GCC visibility push(default)
 
 namespace pulsar {
 
-    class ClientConfiguration;
-    class Authentication;
+class ClientConfiguration;
+class Authentication;
+
+class AuthenticationDataProvider {
+ public:
+  virtual ~AuthenticationDataProvider();
```

Comment:
I'm fine with 2 spaces but I think we should probably keep the current width for now. Because changing indent width makes tons of changes and it can hides other unwanted changes.

Reply:
Same for me. We should maintain 4 spaces for indent. Almost all code is already using that.